smoke-test/src/main/java/org/opennms/smoketest/telemetry/Sender.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2019 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2019 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.smoketest.telemetry;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetSocketAddress;
+import java.util.Objects;
+
+public interface Sender {
+    void send(final byte[] payload) throws IOException;
+
+    static Sender udp(final InetSocketAddress destinationAddress) {
+        Objects.requireNonNull(destinationAddress);
+
+        return payload -> {
+            try (DatagramSocket serverSocket = new DatagramSocket()) {
+                final DatagramPacket sendPacket = new DatagramPacket(payload, payload.length, destinationAddress);
+                serverSocket.send(sendPacket);
+            }
+        };
+    }
+
+    static Sender stream(final OutputStream stream) {
+        return stream::write;
+    }
```

Comment:
We assume non null values, but should probably enforce them either via comment or `Objects.requireNonNull(...)`

Reply:
Enforced via `Objects.requireNonNull` in `udp`. `OutputStream::write` checks anyways.